pandas/tests/test_groupby.py

```diff
@@ -5335,6 +5335,86 @@ def test__cython_agg_general(self):
                 exc.args += ('operation: %s' % op,)
                 raise
 
+    def test_cython_fused_group_algos(self):
+        #GH 4095
+        dtypes = [np.int8, np.int16, np.int32, np.int64,
+                  np.uint8, np.uint32, np.uint64,
+                  np.float32, np.float64]
+        ops = [(pd.algos.group_cumprod, np.cumproduct),
+               (pd.algos.group_cumsum, np.cumsum)]
+
+        for dtype in dtypes:
+            for pd_op, np_op in ops:
+                data = np.array([[1],[2],[3],[4]], dtype=dtype)
+                ans = np.zeros_like(data)
+                accum = np.array([[0]], dtype=dtype)
+                labels = np.array([0,0,0,0], dtype=np.int64)
```

Comment:
need to put some nans in the tests. (which should be skipped). In theory we could have an option to NOT skip nans, but I don't think its very useful.

Also add some non-numeric (to make sure they are raising ValueError?)